src/sentry/sentry_apps/api/endpoints/sentry_app_authorizations.py

```diff
@@ -73,19 +73,22 @@ def post(self, request: Request, installation) -> Response:
                     user=promote_request_api_user(request),
                 ).run()
             else:
-                return Response({"error": "Invalid grant_type"}, status=403)
-        except APIUnauthorized as e:
-            logger.warning(
-                e,
+                raise SentryAppIntegratorError(
+                    message="Invalid grant_type", status_code=status.HTTP_403_FORBIDDEN
+                )
+        except SentryAppIntegratorError as e:
+            logger.info(
+                "sentry-app-authorizations.error-context",
                 exc_info=True,
                 extra={
+                    "error": e,
```

Comment:
will logging the exception give you all of the information inside it? do you need to do `str(e)` or define the `__str__` function?

Reply:
ah TIL, https://docs.python.org/3/library/logging.html#logging.Logger.debug . We dont actually need the error, as long as, `exc_info = True` then we'll include that traceback or we can do `exc_info=e`.